docs/getting-started.md

```diff
@@ -0,0 +1,158 @@
+file that will spin up everything needed for local development.
+
+You can spin up the stack by using the integrated Docker features of your favorite IDE, by using a dedicated Docker UI
+or by executing the command `docker compose up` from within the `stack` folder.
+
+**Important**: When developing locally, the stack should always be running.
+
+### Stack components
+
+#### Keycloak
+
+- **Website**: <https://www.keycloak.org/>
+- **Purpose**: Run a Keycloak instance as a local SSO provider
+
+#### Keycloak Migration
+
+- **Website**: <https://mayope.github.io/keycloakmigration/>
+- **Purpose**: Migration tool to set up the local SSO provider by executing scripts upon startup
+- **Configuration**: All necessary scripts are located in the
+`stack/keycloak/migration` folder.
+Extensions to Keycloak should be implemented through these scripts.
+
+#### PostgreSQL
+
+- **Website**: <https://www.postgresql.org>
+- **Purpose**: Run a PostgreSQL database instance locally
+
+#### pgAdmin
+
+- **Website**: <https://www.pgadmin.org/>
+- **Purpose**: Run a database management UI pre-configured to connect to the local PostgreSQL instance
+
+#### API gateway
+
+- **Website**: <https://refarch.oss.muenchen.de/gateway>
+- **Purpose**: Launch the API gateway of the RefArch in your local environment
+- **Configuration**: The API gateway behaviour can be adjusted by altering the respective environment variables
+defined in `stack/docker-compose.yml` under the key `refarch-gateway.environment`
+
+#### Appswitcher Server
+
+- **Repository**: <https://github.com/it-at-m/appswitcher-server>
+- **Purpose**: Server component required for appswitcher-vue to access local development tools via the frontend UI
+- **Configuration**: Configure the AppSwitcher within the `stack/appswitcher-server`
+folder.
+
+## Backend & EAI
+
+### Prerequisites
+
+- Java 21
+- Maven 3.9.7
+- Docker (for Postgres and Keycloak)
+
+### Configure templates
+
+After getting the templates, you will need to make a few adjustments:
+
+1. Rename the folder you copied (e.g. from `refarch-backend` to `myapp-backend`).
+2. Change the package name from `de.muenchen.refarch` to
+   `de.muenchen.YourPackageName` (It's recommended to use an IDE for this
+   task, as it will automatically update the imports; otherwise, you will
+   need to do this manually).
+3. Inside the `pom.xml`, update the `artifactId`, `groupId`, and `name`
+   fields.
+4. Install required dependencies by executing the command `mvn install`.
+
+Congratulations, the Backend/EAI is now ready to use.
+
+### Profiles
+
+By default, the template supports two Maven profiles:
+
+- `local`: Contains the necessary configurations for running the application
+  locally, including settings for ports, SSO, and the database
+- `no-security`: Configured to operate without security measures for development
+  or testing purposes.
+
+## Frontend & Web Components
+
+### Prerequisites
+
+- Node.js version 20 to 22
```

Comment:
Why both versions and not just 22? This requires us for always supporting both

Reply:
Will discuss this next year.